pkg/daemon/server/service/pipeline_watermark_query.go

```
@@ -56,33 +57,37 @@ func newVertexWatermarkFetcher(pipeline *v1alpha1.Pipeline) *watermarkFetchers {
 		// Can be checked by querying sinkName_SINK in the service
 		if vertex.Sink != nil {
 			toBufferName := v1alpha1.GenerateSinkBufferName(pipeline.Namespace, pipelineName, vertex.Name)
-			fetchWatermark := createWatermarkFetcher(ctx, pipelineName, toBufferName, vertex.Name)
+			fetchWatermark, err := createWatermarkFetcher(ctx, pipelineName, toBufferName, vertex.Name)
+			if err != nil {
+				log.Fatalw("failed to create watermark fetcher", zap.Error(err))
```

Can we return the error all the way up?